uaclient/entitlements/tests/test_livepatch.py

```diff
@@ -51,9 +51,15 @@ def test_contract_status_entitled(self, entitlement):
 
     def test_contract_status_unentitled(self, entitlement):
         """The contract_status returns NONE when entitled is False."""
-        entitlement.cfg.write_cache(
-            "machine-access-livepatch", {"entitlement": {"entitled": False}}
+        token = entitlement.cfg.read_cache("machine-token")
```

Comment:
I haven't fully thought this through, so please take it with a grain of salt, but might these test changes be simpler if you used `entitlement_factory` directly, rather than using the `entitlement` fixture and then modifying it after the fact? (Looking at the `entitlement` fixture, it might make sense to introduce a `livepatch_entitlement_factory` fixture, which sets affordances/directives in the same way as `entitlement`.)

Something like (completely untested):

```py
@pytest.fixture
def livepatch_entitlement_factory(entitlement_factory):

    def factory_func(entitled: bool = True, suites: List[str] = None):
        affordances = {
            "architectures": ["x86_64"],
            "minKernelVersion": "4.4",
            "kernelFlavors": ["generic", "lowlatency"],
            "tier": "stable",
        }
        directives = {"caCerts": "", "remoteServer": "https://alt.livepatch.com"}
        return entitlement_factory(
            LivepatchEntitlement, affordances=affordances, directives=directives,
            entitled=entitled, suites=suites,
        )

    return factory_func

@pytest.fixture
def entitlement(livepatch_entitlement_factory):
    return livepatch_entitlement_factory()
```

and then this test would be something like:

```py
    def test_contract_status_unentitled(self, livepatch_entitlement_factory):
        entitlement = livepatch_entitlement_factory(entitled=False)
        assert ContractStatus.UNENTITLED == entitlement.contract_status()
```

(This would require adding an `entitled` argument to `entitlement_factory` too, and passing it through to `machine_token` in `conftest.py`.)

Reply:
> I haven't fully thought this through, so please take it with a grain of salt, but might these test changes be simpler if you used `entitlement_factory` directly, rather than using the `entitlement` fixture and then modifying it after the fact? (Looking at the `entitlement` fixture, it might make sense to introduce a `livepatch_entitlement_factory` fixture, which sets affordances/directives in the same way as `entitlement`.)
> 
> Something like (completely untested):
> 
> ```python
> @pytest.fixture
> def livepatch_entitlement_factory(entitlement_factory):
> 
>     def factory_func(entitled: bool = True, suites: List[str] = None):
>         affordances = {
>             "architectures": ["x86_64"],
>             "minKernelVersion": "4.4",
>             "kernelFlavors": ["generic", "lowlatency"],
>             "tier": "stable",
>         }
>         directives = {"caCerts": "", "remoteServer": "https://alt.livepatch.com"}
>         return entitlement_factory(
>             LivepatchEntitlement, affordances=affordances, directives=directives,
>             entitled=entitled, suites=suites,
>         )
> 
>     return factory_func
> 
> @pytest.fixture
> def entitlement(livepatch_entitlement_factory):
>     return livepatch_entitlement_factory()
> ```
> 
> and then this test would be something like:
> 
> ```python
>     def test_contract_status_unentitled(self, livepatch_entitlement_factory):
>         entitlement = livepatch_entitlement_factory(entitled=False)
>         assert ContractStatus.UNENTITLED == entitlement.contract_status()
> ```
> 
> (This would require adding an `entitled` argument to `entitlement_factory` too, and passing it through to `machine_token` in `conftest.py`.)

I like this suggestion a lot, that whole logic I added in testing was messy and felt dirty. Thanks for the thoughts on this will implement tomorrow.